### PR TITLE
snap

### PR DESCRIPTION
**Pull request by mikeroyal**
Added snapcraft.yaml to build [snap](https://snapcraft.io/). This PR addresses issue #1831, though, I still need to make a few tweaks.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: gateway
```
Comment (benfrancis) [inline snap/snapcraft.yaml]: webthings-gateway

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+version: git
+summary: Things Gateway.
```
Comment (benfrancis) [inline snap/snapcraft.yaml]: WebThings Gateway

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+description: Web of Things gateway.
```
Comment (benfrancis) [inline snap/snapcraft.yaml]: Web of Things gateway by Mozilla

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    plugs:
+      - desktop
+      - desktop-legacy
+      - home
+      - x11
+      - browser-support
+      - network
+      - gsettings
+      - pulseaudio
+      - opengl
```
Comment (benfrancis) [inline snap/snapcraft.yaml]: What are all these desktop plugs needed for? The gateway is a headless command line application. X11 shouldn't be necessary.

**Comment by benfrancis** [comment]
@popey might also have feedback?

**Comment by popey** [comment]
I couldn't get it to build here, looks like the yaml is unfinished. `node-engine` should be `nodejs-version`, and the `source: dist/linux-unpacked` assumes something exists in that folder, which for a clean install, is empty. @mikeroyal have you tested building this in a clean VM?

**Comment by mrstegeman** [comment]
I had the same question about all of the desktop dependencies.

What is the build process for this?

@benfrancis If the build is straightforward, it shouldn't be difficult to put out a snap release, in the same way we do Docker releases. The main thing we'll be lacking is the extra test time that the regular gateway images get.

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+parts:
+  gateway:
+    plugin: nodejs
+    node-engine: "12.4.0"
```
Comment (mrstegeman) [inline snap/snapcraft.yaml]: None of our add-ons are currently built for Node 12. Our images right now use Node 8, so it'd be great if we could stick with that for the time being. That can be changed when #2113 is resolved.

Comment (mikeroyal) [reply snap/snapcraft.yaml]: I was not sure what of node version was being used so thank you for me know I will that fix :+1:

File: snap/snapcraft.yaml (hunk 4 above)
Comment (mikeroyal) [reply snap/snapcraft.yaml]: Yeah, that's my bad I thought I had removed in a early version. So thank you for catching that I will remove the plus. :+1:

**Comment by mikeroyal** [comment]
hi @popey, I'm still making changes to the .yaml file. So it's still not complete but I will build it in a clean Ubuntu VM when I've made the necessary corrections. :+1:

**Comment by mrstegeman** [comment]
@mikeroyal Curious if you're still working on this?

**Comment by mikeroyal** [comment]
Yes, @mrstegeman I'm still working on this just been busy with my day job. I would be making some updates this week to the snap. :)

**Comment by mrstegeman** [comment]
@mikeroyal Thanks for the update! Looking forward to getting this merged soon.

**Comment by mrstegeman** [comment]
We've got a separate repo going for this now: https://github.com/mozilla-iot/gateway-snap

**Pull request closed without merging**